simde/x86/avx512/fixupimm.h

```diff
@@ -0,0 +1,113 @@
+#if !defined(SIMDE_X86_AVX512_FIXUPIMM_H)
+#define SIMDE_X86_AVX512_FIXUPIMM_H
+
+#include "types.h"
+#include "flushsubnormal.h"
+
+HEDLEY_DIAGNOSTIC_PUSH
+SIMDE_DISABLE_UNWANTED_DIAGNOSTICS
+SIMDE_BEGIN_DECLS_
+
+#if defined(SIMDE_X86_AVX512F_NATIVE) && defined(SIMDE_X86_AVX512VL_NATIVE)
```

Comment:
I notice you've gotten into the habit of defining the function conditionally lately, but that's not what we usually do in SIMDe.  We unconditionally define the function, then define a macro to override the function.  This is mostly because you can do things with a function that you can't with a macro; for example, use them reliably in a CONSTIFY macro.

When the function is nonsense without the supporting macro, like it is with some of those functions where you create an internal function, then wrap it with a macro which handles the immediate-mode parameter, you should only define the function conditionally.  In cases like this, though, please define the function unconditionally.